core/Pos/Binary/Core/Types.hs

```diff
@@ -61,3 +61,32 @@ instance Bi T.SharedSeed where
 instance Bi T.ChainDifficulty where
     get = label "ChainDifficulty" $ T.ChainDifficulty . getUnsignedVarInt <$> get
     put (T.ChainDifficulty c) = put (UnsignedVarInt c)
+
+instance Bi T.BlockVersionData where
+    get = label "BlockVersionData" $ do
```

Comment:
The reason I remember was “we want serialization to be explicit” (like, someone should be able to just look at the code and see exactly how things are being serialized). It doesn't seem to be a bad decision to me – yeah, you might accidentally forget a field, but on the other hand you are protected against such things as “I changed order of fields and got fucked up”. Writing tests that catch the former is easier than writing tests that catch the latter (since we'd have to keep a corpus of data serialized with previous versions of code, or something).

This said, the best solution is probably using something like https://github.com/chpatrick/codec whenever we can.